service/controller/app/v1/resource/chart/update_test.go

```diff
@@ -0,0 +1,148 @@
+			currentResource: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "prometheus",
+					Labels: map[string]string{
+						"app": "prometheus",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.1.tgz",
+				},
+			},
+			desiredResource: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "prometheus",
+					Labels: map[string]string{
+						"app": "prometheus",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.0.tgz",
+				},
+			},
+			expectedChart: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "prometheus",
+					Labels: map[string]string{
+						"app": "prometheus",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.0.tgz",
+				},
+			},
+		},
+		{
+			name: "case 1: chart should not be update",
+			currentResource: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "prometheus",
+					Labels: map[string]string{
+						"app": "prometheus",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.0.tgz",
+				},
+			},
+			desiredResource: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "prometheus",
+					Labels: map[string]string{
+						"app": "prometheus",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.0.tgz",
+				},
+			},
+			expectedChart: &v1alpha1.Chart{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			var err error
+
+			kc, err := kubeconfigtest.New(nil)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+			c := Config{
+				G8sClient:      fake.NewSimpleClientset(),
+				K8sClient:      k8sfake.NewSimpleClientset(),
+				KubeConfig:     kc,
```

Comment:
```suggestion
				KubeConfig:     kubeconfigtest.New(kubeconfigtest.Config{}),
```

As per previous comment this needs to move to the `kubeconfig` repo. You can use this signature which keeps the tests simpler.

See https://github.com/giantswarm/helmclient/blob/c93791a1e9e60a08f2ee48394d416a726c5e00f2/helmclienttest/helmclient.go for an example.

Reply:
We decided that we revert this change to use it previous method.